app/state.go

```diff
@@ -742,6 +742,13 @@ func (u *updateStoreState) Handle(ctx *StateContext, c Controller) (State, bool)
 		u.update.Artifact.PayloadTypes[n] = i.GetType()
 	}
 
+	// For artifact version >= 3 we need to fetch the artifact provides
+	// header-info field.
+	if provides := installer.GetArtifactProvides(); provides != nil {
+		u.update.Artifact.ArtifactName = provides.ArtifactName
+		u.update.Artifact.ArtifactGroup = provides.ArtifactGroup
+	}
```

Comment:
Here you will need to update `ArtifactProvidesBuffer`, because the returned data may contain more entries than just ArtifactName and ArtifactGroup. The `type-info` version of `artifact_provides` is a [free form key/value store](https://github.com/mendersoftware/mender-artifact/blob/master/Documentation/artifact-format-v3.md#artifact_provides-1). The `MergeArtifactProvides` call that I mention elsewhere will give you both.

See also the next comment.